test/commands/code-analyzer/init.test.ts

```diff
@@ -0,0 +1,78 @@
+import {stubSfCommandUx} from '@salesforce/sf-plugins-core';
+import {TestContext} from '@salesforce/core/lib/testSetup';
+import InitCommand from '../../../src/commands/code-analyzer/init';
+import {InitAction} from '../../../src/lib/actions/InitAction';
+
+describe('`code-analyzer init` tests', () => {
+	const $$ = new TestContext();
+	let spy: jest.SpyInstance;
+	let receivedActionInput: object;
+
+	beforeEach(() => {
+		stubSfCommandUx($$.SANDBOX);
+		spy = jest.spyOn(InitAction.prototype, 'execute').mockImplementation((input) => {
+			receivedActionInput = input;
+		});
+	});
+
+	afterEach(() => {
+		jest.restoreAllMocks();
+	})
+
+	describe('--template', () => {
+		it('Accepts the value, "empty"', async () => {
+			const inputValue = "empty";
+			await InitCommand.run(['--template', inputValue]);
+			expect(spy).toHaveBeenCalled();
+			expect(receivedActionInput).toHaveProperty('template', inputValue);
+		});
+
+		it('Accepts the value, "default"', async () => {
+			const inputValue = "default";
+			await InitCommand.run(['--template', inputValue]);
+			expect(spy).toHaveBeenCalled();
+			expect(receivedActionInput).toHaveProperty('template', inputValue);
+		});
+
+		it('Rejects all other values', async () => {
+			const inputValue = "asdf";
+			let errorThrown: boolean;
+			let message: string = '';
+			try {
+				await InitCommand.run(['--template', inputValue]);
+				errorThrown = false;
+			} catch (e) {
+				errorThrown = true;
+				message = e.message;
+			}
+			expect(errorThrown).toBe(true);
+			expect(message).toContain(`Expected --template=${inputValue} to be one of: empty, default`);
```

Comment:
Any reason why you aren't just using expect(() => ...).toThrow(...)  instead of doing the try catch yourself?

Reply:
...Because I forgot that was an option. Good catch.